caffe2/core/tensor_factories.cc

```diff
@@ -0,0 +1,11 @@
+#include "tensor_factories.h"
+
+namespace caffe2 {
+
+Tensor empty(const vector<int64_t>& dims, const at::TensorOptions& options) {
+  auto tensor = Tensor(dims, at::backendToDeviceType(options.backend()));
+  tensor.raw_mutable_data(scalarTypeToTypeMeta(options.dtype()));
```

Comment:
see here :)

Reply:
With, of course, an eventual refactoring to not go through `raw_mutable_data` to trigger the allocation :)